public/apps/configuration/header/header-components.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React from 'react';
+import { flow } from 'lodash';
+import { ControlProps, DescriptionProps, HeaderProps } from './header-props';
+import { getBreadcrumbs } from '../utils/resource-utils';
+
+// controlType should be one of: https://github.com/AMoo-Miki/OpenSearch-Dashboards/blob/header-collective/src/plugins/navigation/public/top_nav_menu/top_nav_control_data.tsx#L91
```

Comment:
Nit: This should be a permalink in case that code branch chanages

Reply:
Removed this comment since that was merged into the main repo.